src/common/iop_order.c

```diff
@@ -1016,6 +1017,53 @@ int dt_ioppr_move_iop_after(GList **_iop_list, dt_iop_module_t *module, dt_iop_m
 //--------------------------------------------------------------------
 // from here just for debug
 //--------------------------------------------------------------------
+int dt_ioppr_check_db_integrity()
+{
+  int ret = 0;
+  sqlite3_stmt *stmt;
+
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT imgid, operation, module FROM main.history WHERE iop_order <= 0 OR iop_order IS NULL",
+                              -1, &stmt, NULL);
+  if(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    ret = 1;
+    fprintf(stderr, "\nThere are unassigned iop_order in the history!!!\n\n");
+    
+    int count = 0;
+    while(sqlite3_step(stmt) == SQLITE_ROW && count++ < 20)
```

Comment:
Here you get the second stmt as the first one is already retrieved from previous sqilte3_step().

Reply:
Right! Fixed.